src/test/java/org/gridsuite/modification/server/utils/MatcherGeneratorModificationInfos.java

```diff
@@ -0,0 +1,90 @@
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.modification.server.utils;
+
+import org.gridsuite.modification.server.dto.GeneratorModificationInfos;
+import org.gridsuite.modification.server.dto.ReactiveCapabilityCurveModificationInfos;
+import org.hamcrest.Description;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Ayoub Labidi <ayoub.labidi at rte-france.com>
+ */
+public class MatcherGeneratorModificationInfos extends MatcherModificationInfos<GeneratorModificationInfos> {
+
+    protected MatcherGeneratorModificationInfos(GeneratorModificationInfos ref) {
+        super(ref);
+    }
+
+    public static MatcherGeneratorModificationInfos createMatcherGeneratorModificationInfos(GeneratorModificationInfos generatorModificationInfos) {
+        return new MatcherGeneratorModificationInfos(generatorModificationInfos);
+    }
+
+    @Override
+    public boolean matchesSafely(GeneratorModificationInfos m) {
+        return super.matchesSafely(m)
+                && m.getEquipmentId().equals(reference.getEquipmentId())
+                && m.getSubstationIds().equals(reference.getSubstationIds())
+                && m.getEquipmentName().equals(reference.getEquipmentName())
+                && Objects.equals(m.getEnergySource(), reference.getEnergySource())
+                && Objects.equals(m.getMinActivePower(), reference.getMinActivePower())
+                && Objects.equals(m.getMaxActivePower(), reference.getMaxActivePower())
+                && Objects.equals(m.getRatedNominalPower(), reference.getRatedNominalPower())
+                && Objects.equals(m.getActivePowerSetpoint(), reference.getActivePowerSetpoint())
+                && Objects.equals(m.getReactivePowerSetpoint(), reference.getReactivePowerSetpoint())
+                && Objects.equals(m.getVoltageRegulationOn(), reference.getVoltageRegulationOn())
+                && Objects.equals(m.getVoltageSetpoint(), reference.getVoltageSetpoint())
+                && Objects.equals(m.getDroop(), reference.getDroop())
+                && Objects.equals(m.getParticipate(), reference.getParticipate())
+                && Objects.equals(m.getRegulatingTerminalId(), reference.getRegulatingTerminalId())
+                && Objects.equals(m.getRegulatingTerminalType(), reference.getRegulatingTerminalType())
+                && Objects.equals(m.getRegulatingTerminalVlId(), reference.getRegulatingTerminalVlId())
+                && Objects.equals(m.getMaximumReactivePower(), reference.getMaximumReactivePower())
+                && Objects.equals(m.getMinimumReactivePower(), reference.getMinimumReactivePower())
+                && Objects.equals(m.getMarginalCost(), reference.getMarginalCost())
+                && Objects.equals(m.getStepUpTransformerReactance(), reference.getStepUpTransformerReactance())
+                && Objects.equals(m.getTransientReactance(), reference.getTransientReactance())
```

Comment:
missing check for:
qPercent
voltageRegulationType

Reply:
Done